src/main/java/io/beanmapper/strategy/MapToRecordStrategy.java

```diff
@@ -168,20 +169,24 @@ private <S> List<Object> getValuesOfFields(final S source, final Map<String, Fie
                 .toList();
     }
 
-    private Object[] getConstructorArgumentsMappedToCorrectTargetType(final Parameter[] parameters, final List<Object> values) {
+    private Object[] getConstructorArgumentsMappedToCorrectTargetType(final Parameter[] parameters,
+                                                                      final Collection<Object> values) {
         Object[] arguments = new Object[parameters.length];
+        Iterator<Object> valueIterator = values.iterator();
         for (var i = 0; i < parameters.length; ++i) {
-            if (i >= values.size() || values.get(i) == null) {
+            Object currentValue = valueIterator.next();
```

Comment:
It looks like this line may throw a NoSuchElementException, due to not checking Iterable#hasNext() before calling Iterable#next().